### ISSUE TITLE
Default value for min-spaces-from-content clashes with prettier

### ISSUE THREAD

**Issue by soenkeliebau** [adrienverge/yamllint#443]
We are using yamllint to lint our pull requests in combination with [Renovate](https://github.com/renovatebot/renovate) to keep dependencies up to date.

Renovate in some scenarios adds inline comments to yaml and separates those from the actual content with one whitespace like so:

```
steps:
      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # renovate: tag=v2.4.0
```

This triggers the comment lint, since the default config for 'min-spaces-from-content' is 2.

I originally created https://github.com/renovatebot/renovate/issues/14093 to fix this, but during the investigation it was discovered that 2 whitespaces clash with [prettier](https://github.com/prettier/prettier), another widely used yaml linter. Prettier simply removes the second whitespace.

Regarding Renovate we have decided to keep the code as is, since yamllint is easy to configure so this warning doesn't show up. 

I mostly wanted to raise this issue as a question on whether there is a specific reason for "2" being the default value here. I have not been able to come up with anything looking like a common best practice so far..

**Comment by adrienverge** [comment]
Hello Sönke,

That's a good question!

The YAML specification doesn't seem to encourage a fixed number of spaces (neither 1 nor 2 nor 3...), so I'd say this is an opiniated arbitrary value. This initial number came from mimicking Python, which is one of the most famous languagues using `#` for comments, and that requires 2 spaces according to PEP8. Since yamllint is written in Python, the choice was made like this. But again, this is arbitrary!

The main outcome of this is (as you already wrote) that yamllint is designed to be customized to fulfill one's needs.

This can be done using either a [configuration file](https://yamllint.readthedocs.io/en/stable/configuration.html), or directly on the command line:
```shell
yamllint -d '{extends: default, rules: {comments: disable}}'
# or:
yamllint -d '{extends: default, rules: {comments: {min-spaces-from-content: 1}}}'
```

> looks like they are aware of this: prettier/prettier#6780
Their stance is: not our problem :)

Yeah, too bad. I share @ssokolow's opinion in https://github.com/prettier/prettier/issues/6780#issuecomment-549024746

**Comment by serv-inc** [comment]
If you try to set this in .pre-commit-config.yaml, the following might work:

```
  - repo: https://github.com/adrienverge/yamllint
    rev: "v1.26.3"
    hooks:
      - id: yamllint
        args: ["-d {extends: default, rules: {comments: {min-spaces-from-content: 1}}}"]
```

**Comment by ssbarnea** [comment]
@adrienverge Would you mind if I would add a new default setting to yamllint that is prettier friendly? I use it a lot and is extremely popular and it would be much easier to configure yamllint to be compatible with it with one liner. I could take care of maintaining that compatibility myself, especially as maintaining ansible-lint is part of my day job and yamllint is one of its dependencies.

Right now we only have `default` and `relaxed` in https://github.com/adrienverge/yamllint/tree/master/yamllint/conf - and I bet that adding `prettier` would prove as a very popular feature if you accept it.

**Comment by adrienverge** [comment]
Hello, the recommended way to use yamllint is with a [configuration file](https://yamllint.readthedocs.io/en/stable/configuration.html) (or conf as cli argument), which solves this problem.

Even with a new set of rules named `prettier`, one would need a configuration file (or conf as cli argument) to use this set. How different would your proposed `prettier` conf be compared to `default` and `relaxed`? Can you paste a complete example of the conf file here?
